PyRoute/DeltaDictionary.py

```diff
@@ -0,0 +1,368 @@
+            if alg_code not in sector.allegiances:
+                sector.allegiances[alg_code] = Allegiance(alg_code, alg_name, base=False, population=alg_race)
+
+        # dig out subsector names, and use them to seed the dict entries
+        sublines = [line for line in headers if '# Subsector ' in line]
+        subsector_names = dict()
+
+        for line in sublines:
+            bitz = line.split(':')
+            alpha = bitz[0][-1]
+            subname = bitz[1].strip()
+            subsector_names[alpha] = subname
+            subsec = SubsectorDictionary(subname, alpha)
+            sector[subname] = subsec
+
+        # now subsectors are seeded, run thru the elements of starlines and deal them out to their respective subsector
+        # dicts
+        dummy = Sector('', '# 0,0')
+        logging.disable(logging.WARNING)
+
+        for line in starlines:
+            # Re-use the existing, battle-tested, validation logic rather than scraping something new and buggy together
+            star = Star.parse_line_into_star(line, dummy, 'scaled', 'scaled')
+            if not star:
+                continue
+            subsec = star.subsector()
+            subname = subsector_names[subsec]
+            sector[subname].items.append(line)
+
+        return sector
+
+    @staticmethod
+    def _partition_file(lines):
+        """
+            Break lines out into headers section, which is retained, and starlines, which gets minimised later on
+            - this assumes downloaded-from-TravellerMap sector file
+        """
+        headers = []
+        starlines = []
+        isheader = True
+        for line in lines:
+            if isheader:
+                headers.append(line)
+                if line.startswith('----'):
+                    isheader = False
+            else:
+                starlines.append(line)
+        return headers, starlines
+
+
+class SubsectorDictionary(dict):
+
+    def __init__(self, name, position):
+        self.name = name
+        self.items = list()
+        self.position = position
+        super().__init__()
+
+    @property
+    def lines(self):
+        return self.items
+
+    @property
+    def skipped(self):
+        return self.items is None
+
+    def __deepcopy__(self, memodict={}):
+        foo = SubsectorDictionary(self.name, self.position)
+        if self.skipped:
+            foo.items = None
+            return foo
+
+        for item in self.items:
+            foo.items.append(copy.deepcopy(item))
+        return foo
+
+    def drop_lines(self, lines_to_drop):
+        foo = SubsectorDictionary(self.name, self.position)
+        if self.skipped:
+            foo.items = None
+            return foo
+
+        nonempty = 0 < len(self.items)
+
+        for item in self.items:
+            if item in lines_to_drop:
+                continue
+            foo.items.append(copy.deepcopy(item))
+
+        if nonempty and 0 == len(foo.items):
```

Comment:
If we started with at least one line, and have dropped all of them, mark the clone as skipped.